sm/op.c

```diff
@@ -0,0 +1,405 @@
+/* -*- C -*- */
```

Comment:
@shashank-parulekar : Can you raise a separate PR for these sm/ changes, which can be reviewed and merged first ?

Reply:
These sm changes are only used by the btree code. If I raise a separate PR then I will have these changes in both the new and the current PR or I will need to discard the current PR and make 2 new ones.